PokeAlarm/Alarms/Telegram/TelegramAlarm.py

```diff
@@ -23,194 +24,270 @@
 
         # Optional Alert Parameters
-        self.__pokemon = self.create_alert_settings(
-            settings.pop('pokemon', {}), self._defaults['pokemon'])
-        self.__pokestop = self.create_alert_settings(
-            settings.pop('pokestop', {}), self._defaults['pokestop'])
-        self.__gym = self.create_alert_settings(
-            settings.pop('gym', {}), self._defaults['gym'])
-        self.__egg = self.create_alert_settings(
-            settings.pop('egg', {}), self._defaults['egg'])
-        self.__raid = self.create_alert_settings(
-            settings.pop('raid', {}), self._defaults['raid'])
-
-        #  Warn user about leftover parameters
-        reject_leftover_parameters(settings, "'Alarm level in Telegram alarm.")
+        alert_defaults = {
+            'bot_token': self._bot_token,
+            'chat_id': self._chat_id,
+            'sticker': self.pop_type(
+                settings, 'sticker', utils.parse_bool, True),
+            'sticker_notify': self.pop_type(
+                settings, 'sticker_notify', utils.parse_bool, False),
+            'message_notify': self.pop_type(
+                settings, 'message_notify', utils.parse_bool, True),
+            'venue': self.pop_type(
+                settings, 'venue', utils.parse_bool, False),
+            'venue_notify': self.pop_type(
+                settings, 'venue_notify', utils.parse_bool, True),
+            'map': self.pop_type(
+                settings, 'map', utils.parse_bool, True),
+            'map_notify': self.pop_type(
+                settings, 'map_notify', utils.parse_bool, False),
+            'max_attempts': self.pop_type(
+                settings, 'max_attempts', int, 3),
+        }
+
+        # Alert Settings
+        self._mon_alert = self.create_alert_settings(
+            'monsters', settings, alert_defaults)
+        self._stop_alert = self.create_alert_settings(
+            'stops', settings, alert_defaults)
+        self._gym_alert = self.create_alert_settings(
+            'gyms', settings, alert_defaults)
+        self._egg_alert = self.create_alert_settings(
+            'eggs', settings, alert_defaults)
+        self._raid_alert = self.create_alert_settings(
+            'raids', settings, alert_defaults)
+
+        # Reject leftover parameters
+        for key in settings:
+            raise ValueError("'{}' is not a recognized parameter for the Alarm"
+                             " level in a Telegram Alarm".format(key))
 
         log.info("Telegram Alarm has been created!")
 
     # (Re)establishes Telegram connection
     def connect(self):
-        self.__client = telepot.Bot(self.__bot_token)
+        pass
+
+    # Set the appropriate settings for each alert
+    def create_alert_settings(self, kind, settings, alert_defaults):
+        default = TelegramAlarm._defaults[kind]
+        default.update(alert_defaults)
+        settings = Alarm.pop_type(settings, kind, dict, {})
+
+        alert = TelegramAlarm.Alert(
```

Comment:
alert = Alert(

Reply:
So far so good, working well with the changes suggested above. Anyway the images quality is not the same as old stickers,, you may let the users pick their own webp URLs or change the images.